### PR TITLE
High card homework for Arthur Adams

### PR DESCRIPTION
**Pull request by arthuradams**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,28 +1,86 @@
...
 var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"];
 var game = {
   deck: [],
   players: [],
   hands: [],
   buildDeck: function(){
-
+      for (var myRank = 0; myRank < values.length; myRank++) {
+        for (var mySuit = 0; mySuit < suits.length; mySuit++) {
+          this.deck.push({ "rank" : values[myRank], "suit" : suits[mySuit]});
```
Comment (nolds9) [inline script.js:10]: This is great, but might also be a useful time to grab the `values`'s index  for each `card` so as to determine a numerical `weight` property

File: script.js [hunk 2]
```diff
@@ -1,28 +1,86 @@
...
+        }
+      }
   },
   shuffleDeck: function(){
+    var currentIndex = this.deck.length, temporaryValue, randomIndex;
+
```
Comment (nolds9) [inline script.js:16]: be sure to site your sources - Fischer-Yates shuffle method

File: script.js [hunk 3]
```diff
@@ -1,28 +1,86 @@
...
+      this.deck[randomIndex] = temporaryValue;
+    }
   },
   getMorePlayers: function(){
-
+    var message = "Enter the name of a new player.";
+    var defaultText = "Name";
+    var name = "";
+    while (this.players.length < 2) {
+      name = prompt(message,defaultText);
```
Comment (nolds9) [inline script.js:35]: could be a good spot to check for bad user input

File: script.js [hunk 4]
```diff
@@ -1,28 +1,86 @@
...
+      this.players.push(name);
+    }
+    message += "\nLeave blank to stop adding players."
+    while (name = prompt(message,"")) {
```
Comment (nolds9) [inline script.js:39]: In general I would try to avoid variable assignment in conditional operators when possible

File: script.js [hunk 5]
```diff
@@ -1,28 +1,86 @@
...
+      else if (suits.indexOf(a.card.suit) < suits.indexOf(b.card.suit)) {
+        return -1;
+      }
+      else if (suits.indexOf(a.card.suit) > suits.indexOf(b.card.suit)) {
+        return 1;
+      }
+      else {
+        return 0;
+      }
+    }
```
Comment (nolds9) [inline script.js:67]: 👍

**Comment by nolds9** [comment]
:+1: Nice work on this assignment. Let me know if you have any questions

File: script.js (hunk 4 above)
Comment (arthuradams) [inline script.js:39]: I appreciate that it's "bad form" and all, but I wanted to see if it worked in JS. And it's **_cool_**. (Yeah, I know, it's cool until someone has to maintain it.)

File: script.js (hunk 2 above)
Comment (arthuradams) [inline script.js:16]: You're absolutely right, and I meant to do that. I used the exact same code for project 1, and I did mention I copied it from Stack Overflow.

**Pull request closed without merging**